src/protocolgame.cpp

```diff
@@ -366,8 +371,13 @@ void ProtocolGame::disconnectClient(const std::string& message) const
 	disconnect();
 }
 
-void ProtocolGame::writeToOutputBuffer(const NetworkMessage& msg)
+void ProtocolGame::writeToOutputBuffer(const NetworkMessage& msg, bool broadcast/* = true*/)
 {
+	if(broadcast && player->isLiveCasting()) {
+		for(auto& spectator : player->spectators) {
+				spectator->writeToOutputBuffer(msg, false);
```

Comment:
I don't think this solution will be accepted https://github.com/otland/forgottenserver/issues/984#issue-42451810 - the work of broadcasting is done in the dispatcher thread.